improve-digital-htb.js

```diff
@@ -237,42 +275,41 @@ function ImproveDigitalHtb(configs) {
          *
          */
 
-         /* ---------- Process adResponse and extract the bids into the bids array ------------*/
+        /* ---------- Process adResponse and extract the bids into the bids array ------------*/
 
-        var bids = adResponse;
+        var bids = adResponse.bid;
 
         /* --------------------------------------------------------------------------------- */
-
         for (var j = 0; j < returnParcels.length; j++) {
             var curReturnParcel = returnParcels[j];
 
             /* ----------- Fill this out to find a matching bid for the current parcel ------------- */
             var curBid;
 
-            for (var i = 0; i < bids.length; i++) {
-
-                /**
-                 * This section maps internal returnParcels and demand returned from the bid request.
-                 * In order to match them correctly, they must be matched via some criteria. This
-                 * is usually some sort of placements or inventory codes. Please replace the someCriteria
-                 * key to a key that represents the placement in the configuration and in the bid responses.
-                 */
-
-                if (curReturnParcel.xSlotRef.someCriteria === bids[i].someCriteria) {
-                    curBid = bids[i];
-                    break;
+            if (bids && bids.length) {
+                for (var i = 0; i < bids.length; i++) {
+
+                    /**
+                     * This section maps internal returnParcels and demand returned from the bid request.
+                     * In order to match them correctly, they must be matched via some criteria. This
+                     * is usually some sort of placements or inventory codes. Please replace the someCriteria
+                     * key to a key that represents the placement in the configuration and in the bid responses.
+                     */
+                    if (curReturnParcel.requestId === bids[i].id) {
+                        curBid = bids[i];
+                        break;
+                    }
                 }
             }
-
             /* ------------------------------------------------------------------------------------*/
 
             /* HeaderStats information */
             var headerStatsInfo = {};
             headerStatsInfo[curReturnParcel.htSlot.getId()] = [curReturnParcel.xSlotName];
 
             /* No matching bid found so its a pass */
-            if (!curBid) {
-                if (__profile.enabledAnalytics.requestTime) {
+            if (!curBid || typeof curBid.price === 'undefined' || curBid.price === 0) {
+                if (__profile.enabledAnalytics.requestTime && __baseClass._emitStatsEvent) {
```

Comment:
It is possible to return JSON where price is 0 or undefined.  This should be interpreted as a no bid, hence the check.  I can either leave this check here or move the check to the loop above where we traverse the bids and set curBid.

In terms of the check on __baseClass._emitStatsEvent, if we don't check, the test code blows up because __baseClass._emitStatsEvent is not defined in this stubbed state.  The following error is produced:

TypeError: __baseClass._emitStatsEvent is not a function